### PR TITLE
docs(validation): Improve process for validation from command line

### PR DESCRIPTION
**Pull request by chriswmackey**
The `validate_model.py` script runs both Pydantic validation and does a re-serialization to Python, which will catch errors beyond those currently in Pydantic (like missing objects that are referenced by name).

@mostaphaRoudsari ,
I just wanted to get your thoughts on including this `validate_model.py`  script, which can be called from command line in order to do a full validation of any model JSON file that people pass in. The ReadMe,MD shows how to use it and I think this will be helpful for @MingboPeng to check whether his C# code is producing the desired results.

### CODE REVIEW

**Comment by chriswmackey** [comment]
Thanks @mostaphaRoudsari .  I will implement a click CLI, then.

**Comment by chriswmackey** [comment]
I added a command line interface with click, using the one for honeybee-core as a template.  The only major difference was that I decided to make click a full dependency of the repo since we are already committing ourselves to exclusive Python3 here with Pydantic. I have verified that I can run the `validate-model` command locally.

Please let me know if this looks correct @mostaphaRoudsari and, if so, I'll implement something similar for honeybee.

File: dragonfly_schema/cli/__init__.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    click.echo('viiiiiiiiiiiiizzzzzzzzz!')
+
+
+@main.command('validate-model')
+@click.argument('model-json')
+def validate_model(model_json):
+    """Validate a Model JSON file against the Honeybee schema.
+    \b
+    Args:
+        model-json: Full path to a Model JSON file.
```
Comment (mostaphaRoudsari) [inline dragonfly_schema/cli/__init__.py]:
```suggestion
        model_json: Full path to a Model JSON file.
```

File: dragonfly_schema/cli/__init__.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import logging
+import json
+
+from dragonfly_schema.model import Model
+
+
+_logger = logging.getLogger(__name__)
+
+
+@click.group(help='Commands for validating JSON files.')
```
Comment (mostaphaRoudsari) [inline dragonfly_schema/cli/__init__.py]: I assume you will probably have commands more than just the validating commands. For instance you might add add a command to translate the schema to an energy model or a radiance model, etc.

Comment (mostaphaRoudsari) [inline dragonfly_schema/cli/__init__.py]: I assume you will probably have commands more than just the validating commands. For instance you might add add a command to translate the schema to an energy model or a radiance model, etc.

File: dragonfly_schema/cli/__init__.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    pass
+
+
+@main.command('viz')
+def viz():
+    """Check if dragonfly-schema is flying!"""
+    click.echo('viiiiiiiiiiiiizzzzzzzzz!')
+
+
+@main.command('validate-model')
```
Comment (mostaphaRoudsari) [inline dragonfly_schema/cli/__init__.py]: I suggest to create a group of commands for `validate` and then add model as a sub-command. That will be easier to extend if one wants to add a new command to validate a face or some other object.

File: dragonfly_schema/cli/__init__.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    """
+    try:
+        assert os.path.isfile(model_json), 'No JSON file found at {}.'.format(model_json)
+    
+        # validate the Model JSON
+        click.echo('Validating Model JSON ...')
+        Model.parse_file(model_json)
+        click.echo('Pydantic validation passed.')
+        try:
+            import dragonfly.model as df_model
```
Comment (mostaphaRoudsari) [inline dragonfly_schema/cli/__init__.py]: Isn't dragonfly a dependency for this module? If it is then you should not need this check.

Comment (mostaphaRoudsari) [reply dragonfly_schema/cli/__init__.py]: I see that it is not a dependency. I think we should make it a dependency at least for the `cli`.

File: dragonfly_schema/cli/__init__.py (hunk 2 above)
Comment (chriswmackey) [reply dragonfly_schema/cli/__init__.py]: Oh. Was that going to be the job of `honeybee-schema` CLI? I thought that was more of a job for `honeybee-energy` or honeybee-radiance` CLI. Right now, I wasn't thinking there'd be anything beyond validation and schema documentation in this repo.

Comment (mostaphaRoudsari) [reply dragonfly_schema/cli/__init__.py]: That's correct but they will extend this command group. This is basically the entry point for all the commands from the extensions too. So I would make it more generic.

**Comment by chriswmackey** [comment]
I just wanted to say that I addressed all of the comments by adding validation to the dragonfly-core CLI:
https://github.com/ladybug-tools/dragonfly-core/pull/30

So I'm simplifying this PR to just delete the validate.py file.

**Pull request merged**